iiod-responder.c

```diff
@@ -376,13 +376,18 @@ static int iiod_enqueue_command(struct iiod_io *writer, uint8_t op,
 
 bool iiod_io_command_is_done(struct iiod_io *io)
 {
-	uint64_t timeout_us = io->timeout_ms * 1000;
+	uint64_t timeout_us;
 	bool done;
 
 	iio_mutex_lock(io->lock);
 
-	done = (io->write_token && iio_task_is_done(io->write_token)) ||
-		(read_counter_us() - io->w_io.start_time > timeout_us);
+	done = io->write_token && iio_task_is_done(io->write_token);
+
+	if (io->timeout_ms) {
+		timeout_us = io->timeout_ms * 1000;
+
+		done |= read_counter_us() - io->w_io.start_time > timeout_us;
```

Comment:
nit: I guess we do not need `|=`?

Reply:
We actually do - check the original code that it replaces and you'll see why.

Reply:
You're right... I was failing to see the first assignment to `done`. But this now raised another (not that important) question to me... I think we could extend the check to be `if (io->timeout_ms && !done)`. IIUC, there's no point in doing anything else if we are already done. That said, we could safely drop the OR...